### PR TITLE
[Ide] Make option deserialization a bit more permissive

### PR DESCRIPTION
**Pull request by Therzok**
Any enum, string or bool value is allowed. Resort to querying FromXElement via reflection on all types and let roslyn bail if anything's wrong

Also, handle a few more test cases

Fixes log spew like
```
ERROR [2018-05-26 02:24:02Z]: Failed to deserialize type: Microsoft.CodeAnalysis.CodeStyle.CodeStyleOption`1[System.String] value: <CodeStyleOption SerializationVersion="1" Type="String" Value="public,private,protected,internal,static,extern,new,virtual,abstract,sealed,override,readonly,unsafe,volatile,async" DiagnosticSeverity="Hidden" />
System.Runtime.Serialization.SerializationException: Serialization error.
  at MonoDevelop.Ide.RoslynServices.Options.MonoDevelopGlobalOptionPersister.Deserialize (System.Object value, System.Type optionType) [0x001b0] in /Users/therzok/Work/md/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices.Options/MonoDevelopGlobalOptionPersister.cs:241
  at MonoDevelop.Ide.RoslynServices.Options.MonoDevelopGlobalOptionPersister.TryFetch (Microsoft.CodeAnalysis.Options.OptionKey optionKey, System.Object& value) [0x0004e] in /Users/therzok/Work/md/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices.Options/MonoDevelopGlobalOptionPersister.cs:104
ERROR [2018-05-26 02:24:02Z]: Failed to deserialize type: Microsoft.CodeAnalysis.CodeStyle.CodeStyleOption`1[Microsoft.CodeAnalysis.CodeStyle.AccessibilityModifiersRequired] value: <CodeStyleOption SerializationVersion="1" Type="Int32" Value="2" DiagnosticSeverity="Hidden" />
System.Runtime.Serialization.SerializationException: Serialization error.
  at MonoDevelop.Ide.RoslynServices.Options.MonoDevelopGlobalOptionPersister.Deserialize (System.Object value, System.Type optionType) [0x001b0] in /Users/therzok/Work/md/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices.Options/MonoDevelopGlobalOptionPersister.cs:241
  at MonoDevelop.Ide.RoslynServices.Options.MonoDevelopGlobalOptionPersister.TryFetch (Microsoft.CodeAnalysis.Options.OptionKey optionKey, System.Object& value) [0x0004e] in /Users/therzok/Work/md/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.RoslynServices.Options/MonoDevelopGlobalOptionPersister.cs:104
```

### CODE REVIEW

**Comment by Therzok** [comment]
Tl;dr roslyn's option persister only seems to handle bool/enum, but CodeStyleOption seems to serialize for string too
